api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1721,75 +1721,76 @@ type AzurePlatformSpec struct {
-	// +immutable
-	VnetName string `json:"vnetName"`
-
-	// VnetID is the ID of an existing VNET to use in creating VMs.
 	// In ARO HCP, this will be the ID of the customer provided VNET.
 	//
 	// Example: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>
 	//
+	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetID is immutable"
-	// +optional
 	// +immutable
+	// +required
 	VnetID string `json:"vnetID,omitempty"`
 
-	// SubnetName is the name of an existing subnet in VnetName / VnetID where Azure CCM will find an existing load
-	// balancer to be used for node egress.
+	// SubnetID is the subnet ID of an existing subnet where the load balancer for node egress will be created. This
+	// subnet is expected to be a subnet within the VNET specified in VnetID. This subnet is expected to exist under the
+	// same subscription as SubscriptionID.
 	//
-	// Example: if your SubnetID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/<subnetName>,
-	//          your SubnetName is <subnetName>
+	// In ARO HCP, managed services will create the aforementioned load balancer in ResourceGroupName.
 	//
-	// +kubebuilder:default:=default
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SubnetID is immutable"
 	// +kubebuilder:validation:Required
-	SubnetName string `json:"subnetName"`
+	// +immutable
+	// +required
+	SubnetID string `json:"subnetID"`
 
 	// SubscriptionID is a unique identifier for an Azure subscription used to manage resources.
 	//
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SubscriptionID is immutable"
 	// +immutable
+	// +required
 	SubscriptionID string `json:"subscriptionID"`
 
 	// MachineIdentityID is used as the user-assigned identity to be assigned to the VMs
 	//
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 
-	// SecurityGroupName is the name of an existing security group on SubnetName. This field is provided as part of the
-	// configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM).
-	//
-	// Example: if your Network Security Group ID is /subscriptions/<subscriptionID>/resourcegroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<securityGroupName>,
-	//          your SecurityGroupName is <securityGroupName>
+	// SecurityGroupID is the ID of an existing security group on the SubnetID. This field is provided as part of the
+	// configuration for the Azure cloud provider, aka Azure cloud controller manager (CCM). This security group is
+	// expected to exist under the same subscription as SubscriptionID and in the same resource group VnetID is located
```

Comment:
I don't think it needs to exist in the same resource group as the VNET now.

Reply:
That's the current assumption atm until we make changes in [HOSTEDCP-1557](https://issues.redhat.com/browse/HOSTEDCP-1557).